daemon/networkdriver/portallocator/portallocator.go

```diff
@@ -59,6 +68,31 @@ func NewErrPortAlreadyAllocated(ip string, port int) ErrPortAlreadyAllocated {
 	}
 }
 
+func init() {
+	const param = "/proc/sys/net/ipv4/ip_local_port_range"
+
+	file, err := os.Open(param)
+	if err != nil {
+		log.Errorf("Failed to read %s kernel parameter: %s", param, err.Error())
```

Comment:
`err` can be transfered to `%s` directly by `fmt` package, so you don't need `Error()`.